File: C++/Problems/Problems/problem_2996.cpp
#include "problem_2996.h"

bool problem_2996::test() {
    std::vector nums  { 1, 2, 3, 2, 5 };

    int expected = 6;

    auto result = missingInteger(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2996::missingInteger(std::vector<int> &nums) {
    int n = nums.size();
    int s = nums[0];
    for (int i = 1; i < n; ++i)
    {
        if (nums[i] != nums[i - 1] + 1)
            break;
        s += nums[i];
    }
    if (s > 50)
        return s;
    std::vector present(51, false);
    for (auto num : nums)
        present[num] = true;
    for (int i = s; i <= 50; ++i)
        if (!present[i])
            return i;
    return 51;
}

File: Python/Problems/Problems/problem2996.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem2996(Problem):
    def test(self) -> bool:
        nums = [1, 2, 3, 2, 5]

        expected = 6

        result = self.missingInteger(nums)

        print(result)

        return result == expected

    def missingInteger(self, nums: List[int]) -> int:
        n = len(nums)
        s = nums[0]
        for i in range(1, n):
            if nums[i] != nums[i - 1] + 1:
                break
            s += nums[i]

        if s > 50:
            return s
        present = [False] * 51
        for num in nums:
            present[num] = True


        for i in range(s, 51):
            if not present[i]:
                return i
        return 51